### PR TITLE
Update cf-1398.mdx  - Py sol

### PR DESCRIPTION
**Pull request by Garlicbread12**
Py sol is very similar to the editorial sol. On the USACO editor, it displays properly, but it doesn't display properly on Github. Please help!

### CODE REVIEW

**Comment by maggie-j-liu** [comment]
as long as it displays properly in the usaco guide editor, it's fine (it doesn't display well in the github preview because github needs a blank line after the pysection tag, but this isn't necessary for the website)

File: solutions/cf-1398C.mdx [hunk 1]
```diff
@@ -99,4 +99,32 @@ public class GoodSubarrays {
 ```
 </JavaSection>
 
+<PySection>
+```py
+import sys
+input = sys.stdin.readline
+print = sys.stdout.write
```
Comment (ryanchou-dev) [inline solutions/cf-1398C.mdx:106]:
```suggestion
```
if this is stdin, then this isn't necessary

File: solutions/cf-1398C.mdx [hunk 2]
```diff
@@ -99,4 +99,32 @@ public class GoodSubarrays {
...
+		prefix_sum += int(nums[j])
+		val = prefix_sum - j - 1 
+		if val not in record:
+			record[val] = 0
+		record[val] += 1
+
+	for id in record:
+	# sum of positive integers less than 'f'
+		ans += (record[id] * (record[id] - 1)) // 2 
+	print(str(ans) + '\n')
```
[the remark this answers is not in the document]
Comment (ryanchou-dev) [reply solutions/cf-1398C.mdx:126]: yeah (remove the `+ '\n'`)

File: solutions/cf-1398C.mdx [hunk 3]
```diff
@@ -99,4 +99,32 @@ public class GoodSubarrays {
 ```
 </JavaSection>
 
+<PySection>
+```py
+import sys
```
[the remark this answers is not in the document]
Comment (ryanchou-dev) [reply solutions/cf-1398C.mdx:104]: we can just delete this, there aren't anymore uses of  `sys` other than input.

File: solutions/cf-1398C.mdx [hunk 4]
```diff
@@ -99,4 +99,32 @@ public class GoodSubarrays {
...
+input = sys.stdin.readline
+print = sys.stdout.write
+# pretty much the official editorial sol
+t_cases = int(input())
+for i in range(t_cases):
+	len_nums = int(input())
+	nums = input().strip()
+	# start with 0 at the head of list
```
Comment (ryanchou-dev) [inline solutions/cf-1398C.mdx:112]:
```suggestion
	# start with 0 at the head of list
```
maybe explain why you start with 0?

Comment (ryanchou-dev) [reply solutions/cf-1398C.mdx:112]: i feel like the line is pretty self-explanatory

File: solutions/cf-1398C.mdx (hunk 2 above)
[the remark this answers is not in the document]
Comment (Garlicbread12) [reply solutions/cf-1398C.mdx:126]: I'm using sys.stdout.write, whereas standard print() is an edited function of the the write function with built-in string formatting. I'll change everything to standard print though.

**Comment by Garlicbread12** [comment]
I removed mentions of sys and used standard input.

**Review by SansPapyrus683 — APPROVED** [review]

File: solutions/cf-1398C.mdx [hunk 5]
```diff
@@ -99,4 +99,28 @@ public class GoodSubarrays {
...
+		prefix_sum += int(nums[j])
+		val = prefix_sum - j - 1 
+		if val not in record:
+			record[val] = 0
+		record[val] += 1
+
+	for key in record:
+	# sum of positive integers less than 'record[key]'
+		ans += (record[key] * (record[key] - 1)) // 2 
+	print(str(ans))
```
Comment (chezbgone) [inline solutions/cf-1398C.mdx:122]: Here `str` is not needed.

**Review by maggie-j-liu — APPROVED** [review]

File: solutions/cf-1398C.mdx [hunk 6]
```diff
@@ -99,4 +99,28 @@ public class GoodSubarrays {
 ```
 </JavaSection>
 
+<PySection>
+```py
+# pretty much the official editorial sol
+t_cases = int(input())
+for i in range(t_cases):
+	len_nums = int(input())
+	nums = input().strip()
```
Comment (Aadit-Ambadkar) [inline solutions/cf-1398C.mdx:108]:
```suggestion
	nums = list(map(int, list(input().strip())))
```

Comment (Aadit-Ambadkar) [reply solutions/cf-1398C.mdx:108]: You will also need to change line 113

File: solutions/cf-1398C.mdx [hunk 7]
```diff
@@ -99,4 +99,28 @@ public class GoodSubarrays {
 ```
 </JavaSection>
 
+<PySection>
+```py
+# pretty much the official editorial sol
+t_cases = int(input())
+for i in range(t_cases):
+	len_nums = int(input())
```
Comment (Aadit-Ambadkar) [inline solutions/cf-1398C.mdx:107]: Change this to `n`

**Review by Aadit-Ambadkar — CHANGES_REQUESTED** [review]

File: solutions/cf-1398C.mdx (hunk 7 above)
Comment (SansPapyrus683) [reply solutions/cf-1398C.mdx:107]: wait bruh why was this accepted

Comment (Garlicbread12) [reply solutions/cf-1398C.mdx:107]: It's fine I changed it.

Comment (SansPapyrus683) [reply solutions/cf-1398C.mdx:107]: yeah sure

File: solutions/cf-1398C.mdx [hunk 8]
```diff
@@ -99,4 +99,28 @@ public class GoodSubarrays {
 ```
 </JavaSection>
 
+<PySection>
```
Comment (ryanchou-dev) [inline solutions/cf-1398C.mdx:102]:
```suggestion
<PySection>

```

**Review by ryanchou-dev — APPROVED** [review]
maybe more comments?

File: solutions/cf-1398C.mdx [hunk 9]
```diff
@@ -99,4 +99,28 @@ public class GoodSubarrays {
...
+```py
+# pretty much the official editorial sol
```
Comment (Aadit-Ambadkar) [inline solutions/cf-1398C.mdx:104]: Not needed

File: solutions/cf-1398C.mdx [hunk 10]
```diff
@@ -99,4 +99,28 @@ public class GoodSubarrays {
...
+n = int(input())
+for i in range(n):
```
Comment (Aadit-Ambadkar) [inline solutions/cf-1398C.mdx:106]: Change these to `t` perhaps? Then instead of `len_nums` use `n`.

**Review by Aadit-Ambadkar — CHANGES_REQUESTED** [review]

File: solutions/cf-1398C.mdx (hunk 10 above)
Comment (SansPapyrus683) [reply solutions/cf-1398C.mdx:106]: agree with the first, disagree with the second

File: solutions/cf-1398C.mdx [hunk 11]
```diff
@@ -99,4 +99,28 @@ public class GoodSubarrays {
 ```
 </JavaSection>
 
+<PySection>
+
+```py
+t = int(input())
+for i in range(t):
```
Comment (SansPapyrus683) [inline solutions/cf-1398C.mdx:106]:
```suggestion
for _ in range(int(input())):
```

**Pull request merged**